scene/animation/animation_node_state_machine.cpp

```diff
@@ -755,7 +773,7 @@ void AnimationNodeStateMachine::remove_node(const StringName &p_name) {
 	emit_signal(SNAME("tree_changed"));
 }
 
-void AnimationNodeStateMachine::rename_node(const StringName &p_name, const StringName &p_new_name) {
+void AnimationNodeStateMachine::rename_node(const StringName &p_name, const StringName &p_new_name, Ref<AnimationNodeStateMachinePlayback> playback) {
```

Comment:
This was strange because you are forcing the user to inform a playback to be able to rename the nodes but it is not possible to guarantee that the informed playback belongs to the current state machine.